source/common/http/http1/codec_impl.cc

```diff
@@ -1086,7 +1086,8 @@ bool ClientConnectionImpl::cannotHaveBody() {
     ASSERT(!pending_response_done_);
     return true;
   } else if (parser_.status_code == 204 || parser_.status_code == 304 ||
-             (parser_.status_code >= 200 && parser_.content_length == 0)) {
+             (parser_.status_code >= 200 && parser_.content_length == 0 &&
+              !(parser_.flags & F_CHUNKED))) {
```

Comment:
Out of curiosity: what coverage do we have for 1xx responses that include content-length or transfer-encoding: chunked headers?